src/persist/persist.ts

```diff
@@ -81,18 +81,14 @@ async function writeFixedChunks(
   mainHeadHash: Hash,
   clientID: string,
 ) {
-  await perdag.withWrite(async dagWrite => {
-    const ps: Promise<void>[] = [];
-    for (const chunk of fixedChunks.values()) {
-      ps.push(dagWrite.putChunk(chunk));
-    }
-    const client: Client = {
-      heartbeatTimestampMs: Date.now(),
-      headHash: mainHeadHash,
-    };
-    // We need to set a head here or the chunks will be GC'd
-    ps.push(setClient(clientID, client, dagWrite));
-    await Promise.all(ps);
-    await dagWrite.commit();
-  });
+  const chunksToPut = fixedChunks.values();
+  await updateClients(clients => {
+    return Promise.resolve({
```

Comment:
or let the function return `MaybePromise<T>`

Reply:
this is a neat one.  thanks for the suggestion.